src/components/ManualAssociateHydrantsForm.js

```diff
@@ -90,15 +94,24 @@ class ManualAssociateHydrantsForm extends React.Component {
       .sort((a, b) => a.get('name') > b.get('name'))
       .map((trail) => {
         const id = trail.get('id');
-        return <MenuItem key={id} value={id}>{trail.get('name')}</MenuItem>;
+        const name = trail.get('name')
+        return { value: id, label: name };
```

Comment:
Put the `toJS()` call at the end of this. Right now it's in the `Select` component's render, so we're calling it for each hydrant instead of just once here.